synfig-studio/src/gui/devicetracker.cpp

```diff
@@ -100,6 +100,8 @@ DeviceTracker::DeviceTracker()
 	list_devices(devices);
```

Comment:
Could you check why does list_devices() adds empty device pointers to the returned array?

Reply:
Since that's a laptop without any extra input device, that could be fingerprint reader as exposed through USB.

Reply:
Shouldn't `list_devices()` be fixed to don't include those empty devices?
Or is it a GTK/gtkmm issue?

Reply:
I mean to erase the null devices before returning from `list_devices()`:
https://github.com/synfig/synfig/blob/3d9c023a08f4d5aff6f129ea283d35cbc3ac9f3d/synfig-studio/src/gui/devicetracker.cpp#L67-L95

This way, any outside code doesn't have to check the null-pointer (and dev no need to know/remember s/he must do it).